in_toto/user_settings.py

```diff
@@ -98,35 +115,93 @@ def get_env():
+    the addition that values that contain colons (:) are split at the position
+    of the colons and converted into a list.
+
+    Section titles in RCfiles are ignored when parsing the key-value pairs.
+    However, there has to be at least one section defined.
+
+    The paths in `RC_PATHS` are ordered in reverse precedence, i.e. each file's
+    settings override a previous file's settings, e.g. a setting defined
+    in `.in_totorc` (in the current working dir) overrides the same
+    setting defined in `~/.in_totorc` (in the user's home dir) and so on ...
+
+    Example:
+
+    ```
+    # E.g. file `.in_totorc` in current working directory
+    [in-toto setting]
+    ARTIFACT_BASE_PATH = /home/user/project
+    ARTIFACT_EXCLUDES = *.link:.gitignore
+    ```
+
+    produces
+
+    ```
+    {
+      "ARTIFACT_BASE_PATH": "/home/user/project"
+      "ARTIFACT_EXCLUDES": ["*.link", ".gitignore"]
+    }
+    ```
 
-  - RCfile format as expected by builtin ConfigParser
-  - The format dictates that there is at least one section,
-    however we are only interested in the values and ignore the sections
-  - Values that contain colons are converted into a list
-  - Variable names are converted to upper case
+  <Exceptions>
+    None.
+
+  <Side Effects>
+    Calls function to read files from disk.
+
+  <Returns>
+    A dictionary containing the parsed key-value pairs.
 
   """
   rc_dict = {}
 
   config = ConfigParser.ConfigParser()
+  # Reset `optionxform`'s default case conversion to enable case-sensitivity
+  config.optionxform = str
   config.read(RC_PATHS)
 
   for section in config.sections():
     for name, value in config.items(section):
-      name = name.upper()
       rc_dict[name] = _colon_split(value)
 
   return rc_dict
 
 
 def set_settings():
-  """Read environment variables and RCfiles and write them to the
-  settings module.
+  """
+  <Purpose>
+    Calls functions that read in-toto related environment variables and RCfiles
+    and overrides variables in `settings.py` with the retrieved values, if they
+    are whitelisted in `IN_TOTO_SETTINGS`.
+
+    Settings defined in RCfiles take precedence over settings defined in
+    environment variables.
+
+  <Exceptions>
+    None.
+
+  <Side Effects>
+    Calls functions that read environment variables and files from disk.
+
+  <Returns>
+    None.
+
   """
   user_settings = get_env()
   user_settings.update(get_rc())
 
-  for name, value in user_settings.iteritems():
-    setattr(SETTINGS_MODULE, name, value)
\ No newline at end of file
+  # If the user has specified one of the settings whitelisted in
+  # IN_TOTO_SETTINGS per envvar or rcfile, override the item in `settings.py`
+  for setting in IN_TOTO_SETTINGS:
+    user_setting = user_settings.get(setting)
+    if user_setting:
+      log.info("Using setting {0}={1}".format(
+          setting, user_setting))
+      setattr(in_toto.settings, setting, user_setting)
```

Comment:
Consider adding an else-statement that logs a debug message for settings that are ignored/invalid.

Reply:
Adding that else-statement would actually log the default settings and not the ignored/invalid settings, which might be more interesting anyway. I will add this.